src/client/mod.rs

```diff
@@ -299,6 +298,39 @@ pub enum QueryResponse {
     GetStoreCost(Result<Token>),
 }
 
+impl QueryResponse {
+    /// Returns true if the result returned is a success or not
+    pub fn is_success(&self) -> bool {
+        use QueryResponse::*;
+        match self {
+            GetBlob(result) => result.is_ok(),
+            GetMap(result) => result.is_ok(),
+            GetMapShell(result) => result.is_ok(),
+            GetMapVersion(result) => result.is_ok(),
```

Comment:
Just thinking if there is a better way to do this? 🤔

Reply:
I assume grouping them with `|` is not allowed by the compiler due to the types?

Reply:
I assumed that as well. I just made the change and it works :) Will update